parlai/core/agents.py

```diff
@@ -816,7 +816,8 @@ def _create_task_agents(opt):
     my_module = importlib.import_module(module_name)
     try:
         # Tries to call the create_agent function in agents.py
-        task_agents = my_module.create_agent(opt)
+        task = sp[0].lower()
```

Comment:
I don't understand this one... what was broken here? Can we leave a comment around that sp[0] magic?

Reply:
I wanted to know the task that is desired to be created so rather than recomputing it in the function i passed it in
NOTE: small thing, i also changed the name to "agents" rather than "agent" because that is more accurate ... a world can actually create multiple agents if it wants

Reply:
I see; I should've read the context above the diff